Core/src/org/sleuthkit/autopsy/report/modules/portablecase/PortableCaseReportModule.java

```diff
@@ -1230,6 +1317,123 @@ private void createAppLaunchBatFile(String destBaseFolder) throws IOException {
+            boolean exists = FileRepository.exists(file);
+            if (exists) {
+                fileRepositoryLookupCache.put(file.getId(), FileRepoState.EXISTS);
+            } else {
+                fileRepositoryLookupCache.put(file.getId(), FileRepoState.DOES_NOT_EXIST);
+            }
+            return exists;
+        } catch (IOException | FileRepositoryException ex) {
+            throw new TskCoreException("Error looking up file in file repository");
+        }
+    }
+    
+    /**
+     * Looks up a list of files in the file repository and caches the results.
+     * The idea is to do the majority of the file lookups in bulk queries.
+     * 
+     * @param contentList   The list of content to check in the file repository.
+     * @param progressPanel the progress panel.
+     * 
+     * @throws TskCoreException 
+     */
+    private void prepopulateFileRepositoryCache(Set<Content> contentList, ReportProgressPanel progressPanel) throws TskCoreException {
+        List<AbstractFile> filesToLookUp = new ArrayList<>();
+        for (Content content : contentList) {
+            
+            if (progressPanel.getStatus() == ReportProgressPanel.ReportStatus.CANCELED) {
+                return;
+            }
+            
+            if (! (content instanceof AbstractFile)) {
+                continue;
+            }
+            AbstractFile file = (AbstractFile)content;
+            
+            // Make sure the hash has been calculated
+            if (file.getSha256Hash() == null || file.getSha256Hash().isEmpty()) {
+                throw new TskCoreException("Missing SHA-256 hash for file " + file.getName() + "(ID: " + file.getId() + ")");
+            }
+            
+            // Check whether we've already looked up this file
+            if (! fileRepositoryLookupCache.containsKey(file.getId())) {
+                filesToLookUp.add(file);
+            }
+            
+            // Look up a batch of files
+            if (filesToLookUp.size() > 200) {
+                doBulkFileRepositoryLookup(filesToLookUp);
+                filesToLookUp.clear();
+            }
+        }
+        
+        // Look up the final batch of files
+        if (! filesToLookUp.isEmpty()) {
+            doBulkFileRepositoryLookup(filesToLookUp);
+            filesToLookUp.clear();
+        }
+    }
+    
+    /**
+     * Look up the batch of files in the file repository and save the returned state
+     * to the cache.
+     * 
+     * @param files File to look up.
+     * 
+     * @throws TskCoreException 
+     */
+    private void doBulkFileRepositoryLookup(List<AbstractFile> files) throws TskCoreException {
+        try {
+            BulkExistenceResult results = FileRepository.exists(files);
+            for (AbstractFile file : files) {
+                BulkExistenceEnum result = results.getResult(file);
+                if (result != null && result.equals(BulkExistenceEnum.TRUE)) {
```

Comment:
Hmm, maybe my BulkExistenceEnum should have EXISTS, DOES_NOT_EXIST, INVALID_HASH, instead of True, False, Invalid.

Reply:
And maybe it should be renamed FileRepoExistenceEnum and FileRepoExistenceResults

Reply:
That might be better. I found the TRUE/FALSE awkward to use